### DISCUSSION TITLE
lazy-load using react-window and react-window-infinite-loader

### DISCUSSION THREAD

**Discussion by ykorach** [TanStack/query#2331] [Q&A]
Hi I am trying to understand what is the best approach using react-query with fetching filtered & sorted data with lazy -load, using virtualized list react-window and react-window-infinite-loader - [sandbox here](https://codesandbox.io/s/5wqo7z2np4)

how can I evaluate isisItemLoaded using react-query in order to load more items if needed?
 and what should be the query keys as the list changed with the filters and sort order?

Thank's!

**Comment by TkDodo** [comment]
you can try `queryClient.getQueryData(keyForThatItem)`. it's only loaded if the data is not `undefined`
